receiver/gitproviderreceiver/internal/scraper/githubscraper/github_scraper.go

```diff
@@ -0,0 +1,70 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package githubscraper // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/gitproviderreceiver/internal/scraper/githubscraper"
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/receiver"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/gitproviderreceiver/internal/metadata"
+)
+
+var (
+	errClientNotInitErr = errors.New("http client not initialized")
+)
+
+type githubScraper struct {
+	client   *http.Client
+	cfg      *Config
+	settings component.TelemetrySettings
+	logger   *zap.Logger
+	mb       *metadata.MetricsBuilder
+}
+
+func (ghs *githubScraper) start(_ context.Context, host component.Host) (err error) {
+	ghs.logger.Sugar().Info("Starting the scraper inside scraper.go")
+	ghs.client, err = ghs.cfg.ToClient(host, ghs.settings)
+	return
+}
+
+func newGitHubScraper(
+	_ context.Context,
+	settings receiver.CreateSettings,
+	cfg *Config,
+) *githubScraper {
+	return &githubScraper{
+		cfg:      cfg,
+		settings: settings.TelemetrySettings,
+		logger:   settings.Logger,
+		mb:       metadata.NewMetricsBuilder(cfg.MetricsBuilderConfig, settings),
+	}
+}
+
+// scrape and return github metrics
+func (ghs *githubScraper) scrape(_ context.Context) (pmetric.Metrics, error) {
+	// TODO: ignoring ctx context.Context for now to pass linting given the removal
+	// of the original code within scraper that leverage the context. Will be added
+	// back in subsequent PRs
+	ghs.logger.Sugar().Debug("checking if client is initialized")
+	if ghs.client == nil {
+		return pmetric.NewMetrics(), errClientNotInitErr
+	}
+
+	now := pcommon.NewTimestampFromTime(time.Now())
+	ghs.logger.Sugar().Debugf("current time: %v", now)
```

Comment:
```suggestion
	ghs.logger.Sugar().Debug("current time", zap.Time("now", now.AsTime()))
```

Same for the other similar log.